### ISSUE TITLE
testsuite failure on powerpc* (ghc 8.0.2)

### ISSUE THREAD

**Issue by LocutusOfBorg** [myfreeweb/http-link-header#6]
Hello, I presume this is a ghc regression, but I would like to understand and ask your opinion.

In the latest ghc transition in Debian, your http-link-header started segfaulting during testsuite compilation on powerpc, ppc64el and ppc64, so I presume we got some powerpc regression somewhere.
log failure doesn't help too much:
https://buildd.debian.org/status/fetch.php?pkg=haskell-http-link-header&arch=ppc64el&ver=1.0.3-3&stamp=1500584246&raw=0
```
[1 of 4] Compiling Network.HTTP.Link.Types ( library/Network/HTTP/Link/Types.hs, dist-ghc/build/Network/HTTP/Link/Types.o )
[2 of 4] Compiling Network.HTTP.Link.Writer ( library/Network/HTTP/Link/Writer.hs, dist-ghc/build/Network/HTTP/Link/Writer.o )
[3 of 4] Compiling Network.HTTP.Link.Parser ( library/Network/HTTP/Link/Parser.hs, dist-ghc/build/Network/HTTP/Link/Parser.o )
[4 of 4] Compiling Network.HTTP.Link ( library/Network/HTTP/Link.hs, dist-ghc/build/Network/HTTP/Link.o )
[1 of 4] Compiling Network.HTTP.Link.Types ( library/Network/HTTP/Link/Types.hs, dist-ghc/build/Network/HTTP/Link/Types.p_o )
[2 of 4] Compiling Network.HTTP.Link.Writer ( library/Network/HTTP/Link/Writer.hs, dist-ghc/build/Network/HTTP/Link/Writer.p_o )
[3 of 4] Compiling Network.HTTP.Link.Parser ( library/Network/HTTP/Link/Parser.hs, dist-ghc/build/Network/HTTP/Link/Parser.p_o )
[4 of 4] Compiling Network.HTTP.Link ( library/Network/HTTP/Link.hs, dist-ghc/build/Network/HTTP/Link.p_o )
Preprocessing test suite 'tests' for http-link-header-1.0.3...
[1 of 4] Compiling Network.HTTP.LinkSpec ( test-suite/Network/HTTP/LinkSpec.hs, dist-ghc/build/tests/tests-tmp/Network/HTTP/LinkSpec.o )
/usr/share/haskell-devscripts/Dh_Haskell.sh: line 1: 16348 Segmentation fault      "$@"
/usr/share/cdbs/1/class/hlibrary.mk:147: recipe for target 'build-ghc-stamp' failed
make: *** [build-ghc-stamp] Error 139
dpkg-buildpackage: error: debian/rules build-arch gave error exit status 2
--------------------------------------------------------------------------------
```

but maybe you have better ideas about how to debug this and to trace it down to forward upstream

thanks!

Just FYI, I already applied 59c4bf5c8b2cffb7b293bbc034310914a6c3bd20 because it was failing on some armhf and probably other platforms too, now we have only that segfault left

**Comment by myfreeweb** [comment]
Nah, I have no idea how to debug this…

**Comment by sjshuck** [comment]
Assuming since there are no further reports that I can close this.

**Comment by LocutusOfBorg** [comment]
FYI
```
  [ Frédéric Bonnard ]
  * debian/rules : disable parallel build on powerpc/ppc64el/ppc64 while
    waiting for fix of https://ghc.haskell.org/trac/ghc/ticket/12537
    (Closes: #876440)
```
